### ISSUE TITLE
No support for Java 7

### ISSUE THREAD

**Issue by nrinaudo** [lucidsoftware/neo-sbt-scalafmt#38]
This is not a huge deal, but `neo-sbt-scalafmt` is compiled for Java 8. I'm typically getting the following error when trying to compile on java 7 (in travis, because who runs java 7 locally):

```
 java.lang.UnsupportedClassVersionError: com/lucidchart/scalafmt/api/ScalafmtFactory : Unsupported major.minor version 52.0
```

I'm wondering whether there's a reason to explicitly exclude java 7 or if that's just an oversight. If the former, it might be good to mention it in the documentation?

It's entirely possible that this _is_ mentioned in the documentation and I did not spot it. If that's the case, I apologise.

**Comment by nrinaudo** [comment]
Bit more feedback on this. The _scala_ code seems to be compiled in the default SBT way:
- when targeting 2.10 or 2.11, the bytecode is java 1.6 compatible
- when targeting 2.12, the bytecode is 1.8 only

The problem seems to be with the _java_ code, which doesn't inherit this behaviour.

I believe this can be addressed by updating the `build.sbt` file:

```scala
lazy val `scalafmt-impl-0.6` = `scalafmt-impl`("0.6.8").settings(
  scalaVersion := "2.11.8",
  // In case of 2.11.8, compile to java 1.6
  javacOptions ++= Seq("-source", "1.6", "-target", "1.6")
)
```

This seems to behave as expected locally, but I'm not 100% of the ramifications this could have. Happy to submit a PR, if you feel it's useful.

As a side note, one of the reasons this matters to me is that I run CI on different JVM versions, and would rather avoid having to set buildfiles up in such a way that the `scalafmt::test` task is only called if running on java 1.8+.

**Comment by nrinaudo** [comment]
After performing more tests, it seems like this might actually be harder to fix than I imagined - `scala.meta` relies on `java.util.Map.putIfAbsent`, which did not exist in the stdlib before 1.8 (there might be other bits, this is just the first I encountered).

So while making `neo-sbt-scalafmt` java 1.6 compatible is simple, it has dependencies that are not themselves 1.6 compatible. This is starting to sound like more trouble than it's worth.
